zhaquirks/tuya/ts110e.py

```diff
@@ -0,0 +1,166 @@
+"""Tuya Dimmer TS110E."""
+from typing import Optional, Union
+
+from zigpy.profiles import zha
+import zigpy.types as t
+from zigpy.zcl import foundation
+from zigpy.zcl.clusters.general import (
+    Basic,
+    GreenPowerProxy,
+    Groups,
+    LevelControl,
+    OnOff,
+    Ota,
+    Scenes,
+    Time,
+)
+
+from zhaquirks.const import (
+    DEVICE_TYPE,
+    ENDPOINTS,
+    INPUT_CLUSTERS,
+    MODELS_INFO,
+    OUTPUT_CLUSTERS,
+    PROFILE_ID,
+)
+
+from zhaquirks.tuya import TuyaDimmerSwitch, TuyaZBExternalSwitchTypeCluster
+
+
+TUYA_LEVEL_ATTRIBUTE = 0xF000
+TUYA_BULB_TYPE_ATTRIBUTE = 0xFC02
+TUYA_MIN_LEVEL_ATTRIBUTE = 0xFC03
+TUYA_MAX_LEVEL_ATTRIBUTE = 0xFC04
+TUYA_CUSTOM_LEVEL_COMMAND = 0x00F0
+
+
+class TuyaLevelPayload(t.Struct):
+    """Tuya Level payload."""
+
+    level: t.uint16_t
+    transtime: t.uint16_t
+
+
+class TuyaBulbType(t.enum8):
+    """Tuya bulb type."""
+
+    LED = 0x00
+    INCANDESCENT = 0x01
+    HALOGEN = 0x02
+
+
+class F000LevelControlCluster(LevelControl):
```

Comment:
This `LevelControl` part is actually making issues. Because it applies to all dimmers in my house now, even the non-Tuya ones. And those obviously don't need this quirk. I think I somehow need to incorporate `CustomCluster` somewhere. But just adding `class F000LevelControlCluster(CustomCluster, LevelControl):` doesn't work.
@javicalle do you know how I can change this quirk to only work with the Custom device `DimmerSwitchWithNeutral1Gang` below?

Reply:
Not sure what you mean. The `` cluster definition is only used in the `DimmerSwitchWithNeutral1Gang` quirk. An this quirk only applies to `_TZ3210_ngqk6jia` device.

What is exactly the problem? When they started? Have you updated HA?

If you believe that there's a problem with your quirk, try to remove it an check if the problem persist, but I don't know how that quirk can affect other devices than the `_TZ3210_ngqk6jia`.

Reply:
So the `zhaquirks.tuya.ts110e.DimmerSwitchWithNeutral1Gang` quirk didn't show up on the new dimmer (non tuya) I bought, but the `F000LevelControlCluster` showed up on the new dimmer I bought. I found that that is because we just used `class F000LevelControlCluster(LevelControl):`, without implementing a `CustomCluster`.
I have experimented some more and found that adding `CustomCluster` as mentioned in my last comment didn't work, because that sent the manufacturer ID to the device and the device didn't recognize that. I found out by continuously comparing the logs between the setting that worked and the one that didn't work.

Long story short, I had to add `NoManufacturerCluster` to make it work. That is only in the later versions of `zha-device-handlers`, so I had to update my entire home-assistance instance. I have done that now and updated my quirk to use `class F000LevelControlCluster(NoManufacturerCluster, LevelControl):`. And that ended up doing what I wanted which is that the `F000LevelControlCluster` only applies to the tuya dimmer and not to others.

I will update this PR accordingly as there are some other changes with the latest updates on the dev branch.